src/wp-includes/class-wp-query.php

```diff
@@ -3092,6 +3092,7 @@ public function get_posts() {
 			);
 
 			$new_request = str_replace( $fields, "{$wpdb->posts}.*", $this->request );
+			$new_request = $wpdb->remove_placeholder_escape( $new_request );
```

Comment:
`wpdb::remove_placeholder_escape()` will call `wpdb::placeholder_escape()`, and that function will add a filter to the `'query'` filter if it isn't already there:

https://github.com/wordpress/wordpress-develop/blob/f371e0ea3007e7fbccca12d38a76b5fa05e13da5/src/wp-includes/class-wpdb.php#L2412-L2439

That might be harmless in the end, but I thought it was worth flagging just because any filter added to `'query'` during the RC phase seemed potentially risky.

Reply:
`wpdb::placeholder_escape` will already have been called to add the placeholder in the first place, when `wpdb::prepare` is called. Wouldn't this filter already be set?

Reply:
> Wouldn't this filter already be set?

Looks like it, I ran this in a private window on the front end and it the filter was set:

```php
add_action( 'init', function() {
	global $wp_filter;
	var_dump( $wp_filter[ 'query' ] );
	exit;
} );
```

Reply:
I was thinking that not all queries might generate a call to `prepare()`. But I guess it's probably reasonable to assume that something will have at some point, as Peter's snippet suggests.

Reply:
Prepare is called within the WP_Query class. 

https://github.com/WordPress/wordpress-develop/blob/70dedb086602287f4e0ee97fbf477adda8310963/src/wp-includes/class-wp-query.php#L1429

https://github.com/WordPress/wordpress-develop/blob/70dedb086602287f4e0ee97fbf477adda8310963/src/wp-includes/class-wp-query.php#L1433

https://github.com/WordPress/wordpress-develop/blob/70dedb086602287f4e0ee97fbf477adda8310963/src/wp-includes/class-wp-query.php#L1558

https://github.com/WordPress/wordpress-develop/blob/70dedb086602287f4e0ee97fbf477adda8310963/src/wp-includes/class-wp-query.php#L1574-L1575

https://github.com/WordPress/wordpress-develop/blob/70dedb086602287f4e0ee97fbf477adda8310963/src/wp-includes/class-wp-query.php#L2062

Not just in context where LIKE is used.